cmd/ferretdb/main.go

```diff
@@ -217,12 +221,108 @@ func main() {
 	case "run":
 		run()
 	case "ping":
-		// TODO https://github.com/FerretDB/FerretDB/issues/4246
+		ping()
 	default:
 		panic("unknown sub-command")
 	}
 }
 
+// ping creates connection to FerretDB instance specified by the flags, and runs `ping` command against it.
+// The check is only executed if --setup-database flag is set.
+func ping() {
+	logger := setupLogger(cli.Log.Format, "")
+	checkFlags(logger)
+
+	l := logger.Sugar()
+
+	if cli.Setup.Database == "" {
+		l.Info("Setup database not specified - skipping ping.")
+		return
+	}
+
+	var pingExecuted bool
+
+	if cli.Listen.Addr != "" {
+		pingExecuted = true
+
+		host, port, err := net.SplitHostPort(cli.Listen.Addr)
+		if err != nil {
+			l.Fatal(err)
+		}
+
+		l.Infof("--listen-addr flag is set. Pinging %s...", cli.Listen.Addr)
```

Comment:
Let's log one info+ message per successful/unsuccessful ping (debug logs are fine). Otherwise, Docker's status and logs will be swarmed by ping logs